File: web/src/app/core/models/department.model.ts
import {Identity} from './identity';

export interface DepartmentModel {
  name: string;
  description: string;
  imageUrl: string;
}

export interface Department extends DepartmentModel, Identity {
  _links: any;
}

export interface RichDepartment extends DepartmentModel, Identity {}

export interface DepartmentForm extends DepartmentModel, Identity {}

File: backend/src/main/kotlin/de/thkoeln/colab/roomsserver/core/models/DepartmentModel.kt
/*
 * Copyright (c) 2020  colab  TH-Köln
 * @author Florian Leonhard
 * All rights reserved
 */

package de.thkoeln.colab.roomsserver.core.models

interface DepartmentModel {
    val name: String
    val description: String
    val imageUrl: String?
}

interface DepartmentIdentity: DepartmentModel, IdentityModel